tests/test_proxy_functional.py

```diff
@@ -881,6 +882,21 @@ async def xtest_proxy_from_env_https_with_auth(
     assert r2.headers["Proxy-Authorization"] == auth.encode()
 
 
+@pytest.mark.xfail
+async def xtest_proxy_https_proxy_headers(proxy_test_server: Any, loop: Any) -> None:
+    sess = aiohttp.ClientSession()
+    proxy = await proxy_test_server()
+    proxy.return_value = {"status": 200, "body": b"1" * (2**20)}
+    url = "https://www.google.com.ua/search?q=aiohttp proxy"
+
+    resp = await sess.get(url, proxy=proxy.url)
```

Comment:
I think it's better to use the `async with` context manager because it ensures that resources will be properly closed after the `async with` block completes, even if an exception occurs. The code will also become more readable, because it does not require explicit invocation of closing methods (`close()`, `release()`, etc.).

Here's what the test implementation might look like:
```py
async def test_proxy_https_proxy_headers(make_proxy_test_server: Any) -> None:
    async with aiohttp.ClientSession() as session:
        async with make_proxy_test_server() as proxy:
            proxy.return_value = {"status": 200, "body": b"1" * (2 ** 20)}
            url = "https://www.google.com.ua/search?q=aiohttp proxy"
            async with session.get(url, proxy=proxy.url) as resp:
                assert resp.proxy_headers == {"proxy-agent": "TestProxy/1.0"}
                assert resp.headers == {}
```

Reply:
Agreed (note you can make code suggestions with the first toolbar button, which can then just be comitted).

Reply:
Sure, thank you, I'll take it note

Reply:
Thanks for the suggestion! Applying it!